### PR TITLE
merge 2023-08-01-preview API version to main repo

### PR DESCRIPTION
**Pull request by agrbhuvnesh**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub)).
  - [x] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by agrbhuvnesh** [comment]
@rkmanda; As discussed during ARM office hrs call, We have 2 RP services (HCI RP, VM self-service RP) sharing same namespace but diff resources. we're working to support directory / folder structure for both subset of resources under HCI namespace, since last API version 2023-09-01-preview has been created for other subset of resources, it is reporting 2 issues:
1. [MISSING_APIS_IN_DEFAULT_TAG](https://github.com/Azure/avocado/blob/master/README.md#MISSING_APIS_IN_DEFAULT_TAG) as one default API version tag doesn't have resources of other RP.
2. Last default API version is used with VM self-service RP which doesn't have resources for HCI RP, and now linter rules couldn't catch existing issues and showing all earlier issues with current API version for HCI.

Please help us to suppress these issues as we've urgent timelines to move this API version to public preview.

**Comment by rkmanda** [comment]
pl address the linter errors mentioned here - https://github.com/Azure/azure-rest-api-specs/pull/26239/checks?check_run_id=18234255294

**Comment by agrbhuvnesh** [comment]
> pl address the linter errors mentioned here - https://github.com/Azure/azure-rest-api-specs/pull/26239/checks?check_run_id=18234255294

@rkmanda : above issues are due to 2nd point, 'Last default API version is used with VM self-service RP which doesn't have resources for HCI RP, and now linter rules couldn't catch existing issues and showing all earlier issues with current API version for HCI.'
All issues reported existed earlier with GA API version 2023-08-01.

**Comment by rkmanda** [comment]
Approving linter errors as they are pre-existing

**Comment by kazrael2119** [comment]
Hi @agrbhuvnesh , I'm from JS SDK team, I have some issues about your pr:

1: azurestackhci was released with apiversion 2023-09-01-preview last month, but your version is 2023-08-01-preview, and you change the default tag in readme.md from 09-01-preview to 08-01-preview. So do you want to release your tag in the future? If so, unfortunately we are not allowed to release the older version preview.

2: please fix the duplicate schema ```Cluster``` in [clusters.json](https://github.com/Azure/azure-rest-api-specs/pull/26239/files#diff-046def37040280aa8f2f3175cc9ab2910ace3c8dcbd5b8eedca449e751da854eR481) and [deploymentSettings.json](https://github.com/Azure/azure-rest-api-specs/pull/26239/files#diff-ada0c3fc4778f63049d3981b7fb9b1db4609b68d982361b46a84e3c505f3b310R297) and ```step``` in [deploymentSettings.json](https://github.com/Azure/azure-rest-api-specs/pull/26239/files#diff-ada0c3fc4778f63049d3981b7fb9b1db4609b68d982361b46a84e3c505f3b310R879) and [updateRuns.json](https://github.com/Azure/azure-rest-api-specs/pull/26239/files#diff-634f2cef9f59295107b49e5e5641c82561376568d47860e7d66cd77b0a37700eR334)

3 @msyyc @Alancere , please help check the ci sdk generation error for python and go, thanks

**Comment by priyjain358** [comment]
Hi @kazrael2119 ,
1. We have 2 RP services (HCI RP, VM self-service RP) sharing same namespace but diff resources. we're working to support directory / folder structure for both subset of resources under HCI namespace. Till then, 2023-09-01-preview was released with one subset of resources that are still in private preview (there is no SDK release for them currently). The second set of resources, ones being released with 2023-08-01-preview are in public preview and there is SDK support for them. We want to change the default tag to 2023-08-01-preview for the SDK generation support itself.

File: specification/azurestackhci/resource-manager/readme.md [hunk 1]
```diff
@@ -85,6 +92,25 @@ input-file:
...
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/extensions.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/offers.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/operations.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/publishers.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/skus.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/updateRuns.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/updateSummaries.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/updates.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/deploymentSettings.json
+  - Microsoft.AzureStackHCI/preview/2023-08-01-preview/edgeDevices.json
```
Comment (qiaozha) [inline specification/azurestackhci/resource-manager/readme.md:112]: @priyjain358  Is your change for this PR  just adding arcSettings, clusters, extensions back and add new resources ? If yes, could you support the 2023-09-01-preview version and add it to the package-preview-2023-09, we've already released SDK with that tag, and are not allowed to release an older version.

Comment (qiaozha) [reply specification/azurestackhci/resource-manager/readme.md:112]: With that, we could just release a minor version to add those back.

Comment (priyjain358) [reply specification/azurestackhci/resource-manager/readme.md:112]: @qiaozha, we didn't raise any SDk release request, when did we release SDK with 2023-09-01-preview version? It's still in private preview..

Comment (priyjain358) [reply specification/azurestackhci/resource-manager/readme.md:112]: We have already released 2023-08-01-preview in private and want to release it as public preview now, hence we can't include those changes in 2023-09-01-preview. 
If we don't change the default tag, what are the implications? Will we be able to release SDK with 08-01-preview?

**Comment by priyjain358** [comment]
@msyyc  @Alancere please see breaking change for python and Go

**Comment by kazrael2119** [comment]
> Hi @kazrael2119 ,
> 
> 1. We have 2 RP services (HCI RP, VM self-service RP) sharing same namespace but diff resources. we're working to support directory / folder structure for both subset of resources under HCI namespace. Till then, 2023-09-01-preview was released with one subset of resources that are still in private preview (there is no SDK release for them currently). The second set of resources, ones being released with 2023-08-01-preview are in public preview and there is SDK support for them. We want to change the default tag to 2023-08-01-preview for the SDK generation support itself.
> 2. The definition for both Cluster and Step is different in the files mentioned.
> 
> And I see that the JS SDK check is passing now..

for 2, could you use different names to distinguish "cluster" and "step" in their respective files due to different definitions?

**Review by rkmanda — APPROVED** [review]

**Pull request merged**